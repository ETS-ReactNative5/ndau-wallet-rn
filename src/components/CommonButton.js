import React, { Component } from 'react';

import { StyleSheet } from 'react-native';
import Button from 'react-native-button';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen';
import cssStyles from '../css/styles';

class CommonButton extends Component {
  onPress() {
    this.props.onPress();
  }
  render() {
    return (
      <Button
        style={styles.text}
        disabledContainerStyle={styles.disabledStyle}
        containerStyle={styles.containerStyle}
        onPress={this.props.onPress}
        {...this.props}
      >
        {this.props.title}
      </Button>
    );
  }
}

var styles = StyleSheet.create({
  text: {
    color: '#ffffff',
    fontSize: 16,
    fontFamily: 'TitilliumWeb-Regular',
    color: '#fff'
  },
  disabledStyle: {
    backgroundColor: '#696969'
  },
  containerStyle: {
    alignContent: 'center',
    justifyContent: 'center',
    borderRadius: 3,
    height: hp('7%'),
    backgroundColor: '#4e957a'
  }
});

export default CommonButton;
