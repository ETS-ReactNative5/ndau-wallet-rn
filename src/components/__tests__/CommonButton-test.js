import { mount } from 'enzyme';
import React from 'react';
import { StyleSheet } from 'react-native';
import CommonButton from '../CommonButton';

import renderer from 'react-test-renderer';

describe('testing CommonButton...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  });
  const navigator = {
    setStyle: () => {}
  };

  beforeEach(() => {});
  it('renders correctly', () => {
    const tree = renderer
      .create(<CommonButton parentStyles={styles} navigator={navigator} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
