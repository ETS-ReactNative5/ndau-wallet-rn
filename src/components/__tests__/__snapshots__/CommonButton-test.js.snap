// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CommonButton... renders correctly 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityTraits="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#4e957a",
      "borderRadius": 3,
      "height": 93.5,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#ffffff",
        "fontFamily": "TitilliumWeb-Regular",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  />
</View>
`;
